modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/DigestAuthenticationTestCase.java

```diff
@@ -63,72 +76,68 @@ public void setEnvironment() throws Exception {
-        apiCreationRequestBean.setTiersCollection(APIMIntegrationConstants.API_TIER.UNLIMITED);
-        apiCreationRequestBean.setProvider(providerName);
-
-        //Add the API to the publisher
-        apiPublisher.addAPI(apiCreationRequestBean);
-
-        //Publish the API
-        APILifeCycleStateRequest updateRequest = new APILifeCycleStateRequest(apiName, providerName,
-                APILifeCycleState.PUBLISHED);
-        apiPublisher.changeAPILifeCycleStatus(updateRequest);
-
-        //Login to the API Store
-        apiStore.login(user.getUserName(), user.getPassword());
-
-        //Add an Application in the Store.
-        apiStore.addApplication("DigestAuthAPP", APIMIntegrationConstants.APPLICATION_TIER.UNLIMITED, "",
-                "Test-Digest-Auth");
-
-        //Subscribe to the new application
-        SubscriptionRequest subscriptionRequest = new SubscriptionRequest(apiName, apiVersion, providerName,
-                "DigestAuthAPP",  APIMIntegrationConstants.API_TIER.UNLIMITED);
+        String endpointType = "secured";
+        String applicationName = "DigestAuthAPP";
+
+        APIRequest apiRequest = new APIRequest(apiName, apiContext, new URL(endpointUrl));
+        apiRequest.setVersion(apiVersion);
+        apiRequest.setProvider(providerName);
+        apiRequest.setTiersCollection(APIMIntegrationConstants.API_TIER.UNLIMITED);
+        apiRequest.setTier(APIMIntegrationConstants.API_TIER.UNLIMITED);
+        apiRequest.setEndpointType(endpointType);
+        apiRequest.setEndpointSecurityAuthType("DIGEST");
+        apiRequest.setEpUsername("DigestAuth");
+        apiRequest.setEpPassword("digest123");
+
+        //add api
+        HttpResponse serviceResponse = restAPIPublisher.addAPI(apiRequest);
+        apiId = serviceResponse.getData();
+
+        //publish the api
+        restAPIPublisher.changeAPILifeCycleStatus(apiId, APILifeCycleAction.PUBLISH.getAction(), null);
+
+        //create an application
+        HttpResponse applicationResponse = restAPIStore.createApplication(applicationName, "Test Application",
+                APIMIntegrationConstants.APPLICATION_TIER.UNLIMITED, ApplicationDTO.TokenTypeEnum.OAUTH);
+        applicationId = applicationResponse.getData();
 
         waitForAPIDeploymentSync(providerName, apiName, apiVersion, APIMIntegrationConstants.IS_API_EXISTS);
 
-        apiStore.subscribe(subscriptionRequest);
+        //Subscribe to the new application
+        restAPIStore.createSubscription(apiId, applicationId, APIMIntegrationConstants.API_TIER.UNLIMITED);
 
-        //Generate a production token and invoke the API
-        APPKeyRequestGenerator generateAppKeyRequest = new APPKeyRequestGenerator("DigestAuthAPP");
-        String responseString = apiStore.generateApplicationKey(generateAppKeyRequest).getData();
-        JSONObject jsonResponse = new JSONObject(responseString);
+        ArrayList grantTypes = new ArrayList();
+        grantTypes.add("client_credentials");
```

Comment:
Shall we use the constant here? APIMIntegrationConstants.GRANT_TYPE.CLIENT_CREDENTIAL

Reply:
Fixed